keyboards/kiwikeebsmacro/kiwikeebsmacro.c

```diff
@@ -0,0 +1,27 @@
+/* Copyright 2020 KiwiKeebs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "kiwikeebsmacro.h"
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_AUDIO_VOL_UP);
+        } else {
+            tap_code(KC_AUDIO_VOL_DOWN);
+        }
+    }
+}
```

Comment:
`_user` functions shouldnt be at the keyboard level, and should be moved to keymap.
```suggestion
```

Reply:
This is a bit of a complicated one.  Since there is no boolean value here to detect if it's been overriden by the user function. 

And is complicated by the fact that if you want configurator support, it would need to be here.

A less than ideal compromise may be to add `__attribute__((weak))` here.

Reply:
I guess to be honest guys I am just a little over my head with the encoder, is there no way to program this using QMK configurator? If not I may have to set it as volume and then give users the files to flash and edit the code themselves?

Reply:
Changed encoder_update_user to encoder_update_kb to comply until the config has encoder support.